core/os/time.cpp

```diff
@@ -137,11 +137,15 @@ VARIANT_ENUM_CAST(Time::Weekday);
 		if (p_datetime.find_char('T') > 0) {                      \
 			has_date = has_time = true;                           \
 			PackedStringArray array = p_datetime.split("T");      \
+			if (array.size() < 2)                                 \
+				return iso8601_fmt_err_val;                       \
```

Comment:
I guess it should print at error message, like other functions do:

```suggestion
			ERR_FAIL_COND_V_MSG(array.size() < 2, err_val, "Invalid ISO-8601 date/time string"); \
```

I would also move the error return value to the macro argument, see #60203 changes for example.